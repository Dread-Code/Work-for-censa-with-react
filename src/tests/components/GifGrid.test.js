import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs'); // para crear mocks 

describe('GifGrid Component', () => {
    
    const category = 'One Punch'

    test('Carga del compoente', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category = { category }/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes', () => {
        
        // crear un mock
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualguiercosa.jpg',
            title: 'cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        });
        
        const wrapper = shallow(<GifGrid category = { category }/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(true); // evalua si el componente existe 
        expect( wrapper.find('GifGridItem').length).toBe( gifs.length); // evalua el componente 
        //con respecto a la data que le llegue

    });

});