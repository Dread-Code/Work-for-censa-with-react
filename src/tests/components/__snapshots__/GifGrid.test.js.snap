// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Component Carga del compoente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
     Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`GifGrid Component debe de mostrar items cuando se cargan imagenes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
     Cargando...
  </p>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="https://localhost/cualguiercosa.jpg"
    />
  </div>
</Fragment>
`;
